### ISSUE TITLE
expo-linear-gradient is crashing the iOS app

### ISSUE THREAD

**Issue by RazaShehryar** [expo/expo#16572]
### Summary

Hi,

I have wrapped my multiple components in the LinearGradient component. However, when I try to use it, for example, with TextInput, it crashes the app. To reproduce the issue, tap on the textinput, then while the keyboard is in the closing state, quickly tap on the textinput again. The app will crash. If I remove the expo-linear-gradient component and repeat the same steps, it doesn't crash.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

44

### Environment

expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 11.6.4
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.14.0 - /usr/local/bin/node
      Yarn: 1.22.17 - ~/.yarn/bin/yarn
      npm: 7.24.2 - ~/Projects/Seven/node_modules/.bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 15.0, DriverKit 20.4, macOS 11.3, tvOS 15.0, watchOS 8.0
      Android SDK:
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7621141
      Xcode: 13.0/13A233 - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ^0.16.18 => 0.16.18 
      babel-preset-expo: 9.0.2 => 9.0.2 
      expo: ^44.0.0 => 44.0.6 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      eas-cli: 0.25.0
      expo-cli: 5.0.3
    Expo Workflow: managed

### Reproducible demo

```
 <KeyboardAvoidingView
        behavior={isIphone ? 'padding' : 'height'}
        style={ScreenStyles.fullFlex}>
        <View style={styles.content}>
          <View style={styles.inputSection}>
            <TextInput
              placeholder="Write your message"
              multiline={true}
              value={message}
              onChangeText={setMessage}
              style={styles.input}
              autoComplete={'name'}
            />
          </View>

           <LinearGradient
            style={ScreenStyles.bottomGradient(insets, { paddingTop: 40 })}
            colors={BOTTOM_GRADIENT_COLORS}>
            <Button
            />
          </LinearGradient>
        </View>
      </KeyboardAvoidingView>
```

**Comment by byCedric** [comment]
Hi @RazaShehryar! Thanks for writing in, unfortunately I can't run your repro. It's missing a lot of info, like:
- styles of `ScreenStyles.fullFlex`
- styles of `styles.content`
- styles of `styles.inputSection`
- styles of `styles.input`
- styles of `ScreenStyles.borromGradient()`
- value of `BOTTOM_GRADIENT_COLORS`

Can you provide a full repro for us to test with?

**Comment by RazaShehryar** [comment]
I will just paste the inline styling here.

```

<KeyboardAvoidingView
        behavior={isIphone ? 'padding' : 'height'}
        style={{ flex: 1 }}>
        <View style={{ paddingHorizontal: 15,
    flex: 1,}}>
          <View style={{flexGrow: 1,
    flexShrink: 1,
    maxHeight: 500,
    marginTop: 20,
    marginBottom: 150,}}>
            <TextInput
              placeholder="Write your message"
              multiline={true}
              value={message}
              onChangeText={setMessage}
              style={{ flexBasis: 150,
    flex: 1,
    flexShrink: 0,
    marginTop: 10,}}
              autoComplete={'name'}
            />
          </View>

           <LinearGradient
            style={{alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      left: 0,
      right: 0,
      bottom: 0,
      padding:15,
      }}
            colors={[ hexToRgba(Colors.backgroundGrayDark, 0.0),
  hexToRgba(Colors.backgroundGrayDark, 0.8),
  hexToRgba(Colors.backgroundGrayLight, 1.0),
  hexToRgba(Colors.backgroundGrayLight, 1.0),
  hexToRgba(Colors.backgroundGrayLight, 1.0),]}>
            <Button
            />
          </LinearGradient>
        </View>
      </KeyboardAvoidingView>
```

**Comment by mikgross** [comment]
Hey @byCedric; We are also experiencing LinearGradient crashes on iOS. I have an error report from sentry available and the specific error is:
````
EXC_BREAKPOINT: Fatal error: Use of unimplemented initializer 'init(layer:)' for class 'ExpoLinearGradient.LinearGradientLayer'
````
Do we have any idea of what might happen?

**Comment by RazaShehryar** [comment]
No idea. I got this log from Xcode and this is specifically when the screen is wrapped in the LinearGradient and I try to navigate to other screen quickly. Other than that, it happens when a button is wrapped inside it and I try to tap quickly on the input field

**Comment by turtlefung** [comment]
same, with expo-linear-gradient and textinput, it will crash
After removing the expo-linear-gradient, everything works fine

**Comment by garviand** [comment]
I've been trying to figure out for weeks what was causing crashes on iOS and finally got this log:

```
Process:             Expo Go [16869]
Path:                /Users/USER/Library/Developer/CoreSimulator/Devices/74E6FDAD-F4BC-4245-B5CE-A3E428D7517B/data/Containers/Bundle/Application/97419289-15EF-42B5-835B-84CAD5A0875A/Exponent-2.23.2.tar.app/Expo Go
Identifier:          host.exp.Exponent
Version:             2.23.2 (2.23.2.101)
Code Type:           X86-64 (Native)
Role:                Foreground
Parent Process:      launchd_sim [16659]
Coalition:           com.apple.CoreSimulator.SimDevice.74E6FDAD-F4BC-4245-B5CE-A3E428D7517B [3467]
Responsible Process: SimulatorTrampoline [3108]

Date/Time:           2022-04-18 14:29:15.3979 -0500
Launch Time:         2022-04-18 14:15:52.9109 -0500
OS Version:          macOS 12.3.1 (21E258)
Release Type:        User
Report Version:      104

Exception Type:  EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes: 0x0000000000000001, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: SIGNAL 4 Illegal instruction: 4
Terminating Process: exc handler [16869]

Triggered by Thread:  0

Application Specific Information:
CoreSimulator 802.6 - Device: iPhone 12 Pro Max (74E6FDAD-F4BC-4245-B5CE-A3E428D7517B) - Runtime: iOS 14.5 (18E182) - DeviceType: iPhone 12 Pro Max
Fatal error: Use of unimplemented initializer 'init(layer:)' for class 'ABI44_0_0ExpoLinearGradient.LinearGradientLayer'
dyld4 config: DYLD_ROOT_PATH=/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS 14.5.simruntime/Contents/Resources/RuntimeRoot
```

**Comment by tsapeta** [comment]
It's been fixed by #15843. You can already use the latest version of `expo-linear-gradient` if you use bare workflow, but for Expo Go you'll need to wait for SDK45 that should be released at the beginning of May.
